File: 2024/day03/part1.js
const fs = require('fs')

const input = fs.readFileSync('./input.txt', 'utf8')

const matches = input.matchAll(/mul\([^()]*?\)/gm)

let ans = 0
let i = 0

const a =[...matches]

a.forEach(element => {
	element = element[0].substring(3).replace("(", "").replace(")", "")
	const [a, b] = element.split(",")

	console.log(a, b)

	if (Number.isNaN(parseInt(a)) || Number.isNaN(parseInt(b))) return

	ans += parseInt(a) * parseInt(b)
});

console.log(ans)

File: 2024/day03/part1.py
with open("./input.txt", "r") as file:
	lines = [line.strip() for line in file.readlines()]

line = "".join(lines).split("mul")
print(line)
ans = 0

for pair in line:
	if "(" not in pair and ")" not in pair:
		continue

	cancel = False

	a = ""
	b = ""

	for char in pair[1:]:
		if char == ",":
			break

		a += char

	if not a.isdigit():
		continue

	for char in pair[len(a)+2:]:
		if char == ")":
			break

		if not char.isdigit():
			cancel = True
			break

		b += char
		
	if cancel:
		continue

	ans += int(a) * int(b)

print(ans)